docs/contributing.rst

```diff
@@ -81,47 +81,73 @@ Thank you for your help!
 Running the compiler tests
 ==========================
 
-The ``./scripts/tests.sh`` script executes most Solidity tests automatically,
-but for quicker feedback, you might want to run specific tests.
+Prerequisites
+-------------
+
+Some tests require the `evmone <https://github.com/ethereum/evmone/releases>`_
+library, others require `libz3 <https://github.com/Z3Prover/z3>`_. The test script
+tries to discover the location of the ``evmone`` library, which can be located in
+the current directory, or the ``deps`` folder in the project top level. The required
+file is called ``libevmone.so`` on Linux systems, ``evmone.dll`` on Windows systems
+and ``libevmone.dylib`` on macOS.
+
+Running the tests
+-----------------
 
 Solidity includes different types of tests, most of them bundled into the
 `Boost C++ Test Framework <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/index.html>`_ application ``soltest``.
 Running ``build/test/soltest`` or its wrapper ``scripts/soltest.sh`` is sufficient for most changes.
 
-Some tests require the ``evmone`` library, others require ``libz3``.
+The ``./scripts/tests.sh`` script executes most Solidity tests automatically,
+including those bundled into the `Boost C++ Test Framework <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/index.html>`_ application ``soltest`` (or its wrapper ``scripts/soltest.sh``),
+as well as command line tests and compilation tests.
 
-The test system will automatically try to discover the location of the ``evmone`` library
+The test system automatically tries try to discover the location of the ``evmone`` library
 starting from the current directory. The required file is called ``libevmone.so`` on Linux systems,
 ``evmone.dll`` on Windows systems and ``libevmone.dylib`` on MacOS. If it is not found, the relevant tests
 are skipped. To run all tests, download the library from
 `Github <https://github.com/ethereum/evmone/releases/tag/v0.3.0>`_
 and either place it in the project root path or inside the ``deps`` folder.
 
-If you do not have libz3 installed on your system, you should disable the SMT tests:
-``./scripts/soltest.sh --no-smt``.
+If the ``evmone`` library is not found, tests that use it are skipped. These tests
```

Comment:
This sentence is a duplicate of `If it is not found, the relevant tests ￼are skipped.` a few lines above.

Reply:
I had a merge conflict before last push, so maybe I missed something in the tidy up

Reply:
Actually, this isn't duplication, the first paragraph says how to get evmone, the other says what tests are skipped if it can't be found. To make it clearer, I merged the paragraphs